### PR TITLE
[FIXED JENKINS-30574] Support global default authorization strategy in Authorize Project

### PR DESCRIPTION
**Pull request by stephenc**
Extracted from #13

### CODE REVIEW

**Comment by stephenc** [comment]
@reviewbybees

**Comment by stephenc** [comment]
Depends on #16 as the suggested default strategy for an initial configuration should correspond to a no-op

File: src/main/resources/org/jenkinsci/plugins/authorizeproject/form/dropdownDescriptorSelector.jelly [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+THE SOFTWARE.
+-->
+<?jelly escape-by-default='true'?>
+<j:jelly xmlns:j="jelly:core" xmlns:st="jelly:stapler" xmlns:d="jelly:define" xmlns:l="/lib/layout"
+         xmlns:t="/lib/hudson" xmlns:f="/lib/form">
+  <st:documentation>
+    Renders a single &lt;select> control for choosing a Describable.
+    Depending on the currently selected value, its config.jelly will be
+    rendered below &lt;select, allowing the user to configure Describable.
```
Comment (ikedam) [inline src/main/resources/org/jenkinsci/plugins/authorizeproject/form/dropdownDescriptorSelector.jelly]: Please add a note about the difference from the built-in dropdownDesceiptorSelector, and can be removed with Jenkins 1.645+.

File: src/main/java/org/jenkinsci/plugins/authorizeproject/GlobalQueueItemAuthenticator.java [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+         * {@inheritDoc}
+         */
+        @Override
+        public String getDisplayName() {
+            return Messages.GlobalQueueItemAuthenticator_DisplayName();
+        }
+
+        public AuthorizeProjectStrategy getDefaultStrategy() {
+            return new SystemAuthorizationStrategy();
+        }
```
Comment (ikedam) [inline src/main/java/org/jenkinsci/plugins/authorizeproject/GlobalQueueItemAuthenticator.java]: -1 for defaulting to `SystemAuthorizationStrategy`.
I don't think we have to define a default.

Comment (stephenc) [inline src/main/java/org/jenkinsci/plugins/authorizeproject/GlobalQueueItemAuthenticator.java]: If we don't provide a default then it defaults to "Run as specific user"... and that somehow lets you specify an empty user... so you can very easily end up with
screen shot 2016-02-08 at 12 00 44 and then it's just a short hop, skip and jump to 
screen shot 2016-02-08 at 12 09 28

So I think we do need to specify a default... however thinking some more I can accept a default of `AnonymousAuthorizationStrategy` as that fits with secure by default mode... in other words, the user adding this needs to take a deliberate decision to use the `SystemAuthorizationStrategy`

**Comment by ikedam** [comment]
The described exception caused for
```
Caused by: org.kohsuke.stapler.NoStaplerConstructorException: There's no @DataBoundConstructor on any constructor of class org.jenkinsci.plugins.authorizeproject.strategy.SpecificUsersAuthorizationStrategy
	at org.kohsuke.stapler.ClassDescriptor.loadConstructorParamNames(ClassDescriptor.java:176)
	at org.kohsuke.stapler.RequestImpl$TypePair.convertJSON(RequestImpl.java:576)
```

I'll make a pull request for additional changes for that.

**Pull request merged**